lcc_lokavaluto_app_connection/models/wallet.py

```diff
@@ -136,13 +136,12 @@ def _compute_is_reconversion_allowed(self):
                 record.is_reconversion_allowed = True
 
     def _compute_is_topup_allowed(self):
-        all_rules = self.env["topup.rule"].search([("active", "=", True)])
+        all_rules = self.env["topup.rule"].search([("active", "=", True)], order="sequence")
         for record in self:
-            # By default, topup is not allowed
-            record.is_topup_allowed = False
-            if any(
-                self.search(safe_eval(rule.wallet_domain) + [("id", "=", record.id)])
-                and rule.is_topup_allowed
-                for rule in all_rules
-            ):
-                record.is_topup_allowed = True
+            # By default, topup is allowed
+            record.is_topup_allowed = True
```

Comment:
This will trigger complex python set attribute mechanism, it would be better to only call it if necessary (so preferably after resolution). No idea if checking past value would even be better performance wise.